server/models/Member.js

```diff
@@ -143,7 +145,32 @@ function defineModel() {
     if (member.tier.interval === 'year' && days(new Date(member.lastDonation)) <= 365) {
       return true;
     }
-    return false;
+
+    return models.Order.findOne({
+      where: {
+        CollectiveId: member.tier.CollectiveId,
+        TierId: member.tier.id,
+      },
+    }).then(order => {
+      if (!order) {
+        return false;
+      }
+      return models.Transaction.findOne({
+        where: { OrderId: order.id, CollectiveId: member.tier.CollectiveId },
+        order: [['createdAt', 'DESC']],
+      }).then(transaction => {
+        if (!transaction) {
+          return false;
+        }
+        if (order.interval === 'month' && days(transaction.createdAt) <= 31) {
+          return true;
+        }
+        if (order.interval === 'year' && days(transaction.createdAt) <= 365) {
+          return true;
+        }
+        return false;
+      });
+    });
```

Comment:
This will break [the place](https://github.com/opencollective/opencollective-api/blob/483a5ab7f182d419fb3f87772c7af46c3c87e7e7/server/graphql/v1/queries.js#L805) where this is used since we're now returning a promise instead of a boolean without awaiting it. Let's keep that simple and treat flexible as yearly there: members are active when they have contributed in the past 365 days.

Reply:
Much simpler 👍🏻